xdsl/parser/base_parser.py

```diff
@@ -359,6 +359,36 @@ def parse_integer(
             "Expected integer literal" + context_msg,
         )
 
+    def parse_optional_float(
+        self,
+        *,
+        allow_negative: bool = False,
```

Comment:
I can imagine this being quite confusing, let's drop the default argument for now?

Reply:
It was copied off the above function for integers I think, but sounds like a good idea

Reply:
It seems I copied poorly and the rest have a default value of True, would that be preferred or still no default?

Reply:
I have a slight preference towards default True